packages/yuudachi/src/commands/moderation/anti-raid-nuke.ts

```diff
@@ -39,250 +36,34 @@ export default class implements Command {
 			throw new Error(i18next.t('common.errors.no_mod_log_channel', { lng: locale }));
 		}
 
-		const parsedJoin = ms(args.join);
-		if (parsedJoin < 6000 || isNaN(parsedJoin)) {
-			throw new Error(i18next.t('command.common.errors.duration_format', { lng: locale }));
-		}
-
-		const parsedAge = ms(args.age);
-		if (parsedAge < 6000 || isNaN(parsedAge)) {
-			throw new Error(i18next.t('command.common.errors.duration_format', { lng: locale }));
+		const archiveChannel = await checkLogChannel(
+			interaction.guild,
+			(await getGuildSetting(interaction.guildId, SettingsKeys.GeneralLogChannelId)) as string,
+		);
+		if (!archiveChannel) {
+			throw new Error(i18next.t('common.errors.no_general_log_channel', { lng: locale }));
 		}
 
-		if (args.reason && args.reason.length >= 500) {
-			throw new Error(i18next.t('command.mod.common.errors.max_length_reason', { lng: locale }));
-		}
+		const ignoreRolesId = (await getGuildSetting(interaction.guildId, SettingsKeys.AutomodIgnoreRoles)) as string[];
 
-		const joinCutoff = Date.now() - parsedJoin;
-		const accountCutoff = Date.now() - parsedAge;
+		const sponsorRoleId = (await getGuildSetting(interaction.guildId, SettingsKeys.SponsorRoleId)) as string | null;
+		if (sponsorRoleId) ignoreRolesId.push(sponsorRoleId);
 
-		const fetchedMembers = await interaction.guild.members.fetch({ force: true });
-		const members = fetchedMembers.filter((member) => {
-			if (args.pattern) {
-				try {
-					const re = new RE2(`^${args.pattern}$`, 'i');
-					if (!re.test(member.user.username)) {
-						return false;
-					}
-				} catch {}
+		switch (Object.keys(args)[0]) {
+			case 'file': {
+				return file(interaction, args.file, logChannel, ignoreRolesId, locale, this.redis);
```

Comment:
You shouldn't be passing the redis instance into the function here, but rather resolve it inside of each function by grabbing it from the container.

Reply:
Redis is now singled called from the `executeNuke` function as per [3efc23c](https://github.com/Naval-Base/yuudachi/pull/1069/commits/3efc23c48832b47a932d4eaf076d43a69ea1d958)